setup.py

```diff
@@ -166,6 +166,42 @@ def _is_build_command(argv=sys.argv, cmds=('install', 'build', 'bdist')):
             return arg
 
 
+def _strip_comments(l):
```

Comment:
are these changes a must have?

Reply:
These changes are at least partially needed to include the `dill` library. I used the `setup.py` of the `celery` project to make this. There may not be a need to this, I thought it best to make something coherent with such a closely related project. I am open to discussion though.

Reply:
I am with @auvipy here.
1. I’m not sure this is required, but if so, why?
2. If it is required, this should be done using a separate PR.

Wasn't changing the `default.txt` enough?

Reply:
I think this change brings to light a design flaw with the PR.
I understand using pickle is limiting, but it appears to be a reason why there weren't any requirement.txt files, ever (except for the tests).
Adding a dependency (for `dill` in this case) might be “against the design” of billiard, which is why there wasn’t any support for new dependencies in the setup.py in the first place.

Reply:
> it appears to be a reason why there weren't any requirement.txt files, ever (except for the tests)

What would be the reason? I must say that it isn't obvious from the outside. Would it be that it would add a lot of lines to the `setup.py`? I think not for it doesn't seem a valid reason, unless I am missing something.

> I think this change brings to light a design flaw with the PR

I must say I don't understand why adding a dependency would be a design flaw here, especially as it is designed to perfectly replace the currently used one, only adding in capabilities. Again, this would open up the possibility to pass many unsupported types of objects as parameters (while I have only encountered the `billiard` library when using `celery`, I expect there are other places it is used, and I cannot imagine a situation where it would be detrimental to add these capabilities).

> Adding a dependency (for dill in this case) might be “against the design” of billiard, which is why there wasn’t any support for new dependencies in the setup.py in the first place.

If there is a philosophical issue that I am not aware of in addition to the dependency introduction, I can totally accept it, but regarding this one, it does not seem (to me) that big of a deal. I am well aware that this is ultimately not my decision to make, but who would be in charge of these discussions?

Reply:
Is this resolved?